File: Java/Factorials.java
public class Factorials
{
    // Number of result columns
    final public static int COLUMNS = 4;

    // Limit number for the results
    final public static int LIMIT = 999999999;

    // Result output format
    final public static String RESULT_FORMAT = "%9d";

    // Formatter for the output of results in columns
    private Columns output;


    public static void main(String[] arg_values)
    {
        Factorials factInstance = new Factorials();
        System.out.println("factorials:");
        factInstance.calculate();

        System.out.println();
    }

    public Factorials()
    {
        output = new Columns(COLUMNS, RESULT_FORMAT);
    }

    public void calculate()
    {
        int factor = 1;

        for (int result = 1; Factorials.LIMIT / result >= factor; ++factor)
        {
            result *= factor;
            output.printColumn(result);
        }
        output.finishColumn();
    }
}

File: Python/factorials.py
#!/usr/bin/python

import sys
import columns

# Number of result columns
COLUMNS = 4

# Limit number for the results
LIMIT = 999999999

# Result output format
RESULT_FORMAT = "%9d"


def main():
    sys.stdout.write("factorials:\n")
    factorials()


def factorials():
    factor = 1
    
    result_counter = 0
    result = 1
    while LIMIT / result > factor:
        result *= factor
        result_counter = columns.print_column(COLUMNS, RESULT_FORMAT,
                                              result, result_counter)
        factor += 1
    columns.finish_column(COLUMNS, result_counter)

main()

